### PR TITLE
don't attempt a send when filtering didn't leave any datapoints

### PR DESCRIPTION
**Pull request by petergardfjall**
I discovered that if there were no datapoints left after applying the filters, the send would fail with an error message similar to:

```
time="2018-06-01T11:42:16Z" level=error msg="response from kairosdb map[errors:[Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1]]"
time="2018-06-01T11:42:16Z" level=error msg="req to kairosdb null"
time="2018-06-01T11:42:16Z" level=error msg="failed writing metrics to downstream. error: number of failed datapoints [1] is greater than total datapoints [0]"
```

So when there are no datapoints to send, we simply skip it.

### CODE REVIEW

File: kairosdb/client.go [hunk 1]
```diff
@@ -86,13 +88,19 @@ func NewClient(cfg *config.Config) *Client {
 
 // Send - Apply RelabelConfigs, massage the data and write the samples to KairosDB
 func (c *Client) Send(samples model.Samples) (err error) {
+	logrus.Infof("datapoints prior to filtering: %d", len(samples))
```
Comment (rajatjindal) [inline kairosdb/client.go]: thanks for the PR.

can you please change this loglevel to debug

File: kairosdb/client.go [hunk 2]
```diff
@@ -86,13 +88,19 @@ func NewClient(cfg *config.Config) *Client {
...
 	datapoints := FilterAndProcessSamples(samples, c.cfg)
+	logrus.Infof("datapoints after filtering: %d", len(datapoints))
```
Comment (rajatjindal) [inline kairosdb/client.go]: this one too

**Comment by rajatjindal** [comment]
@johngmyers can you take a peek as well please. looks straight forward fix.

File: kairosdb/client.go [hunk 3]
```diff
@@ -86,13 +88,19 @@ func NewClient(cfg *config.Config) *Client {
...
 
 	filteredSamplesCount := len(samples) - len(datapoints)
 	filteredSamples.WithLabelValues(c.name()).Add(float64(filteredSamplesCount))
 
-	begin := time.Now()
+	if len(datapoints) == 0 {
+		logrus.Infof("empty set of datapoints after filtering, nothing to send.")
```
Comment (johngmyers) [inline kairosdb/client.go]: nit: Should be a semicolon, not a comma splice.

Comment (johngmyers) [reply kairosdb/client.go]: I could be convinced otherwise, but am thinking this should be debug level as well.

**Comment by petergardfjall** [comment]
Updated PR according to comments.

**Comment by johngmyers** [comment]
Finally got public Docker image builds enabled. Built as image tag "0.0.3".

**Pull request merged**